src/Symfony/Bridge/Twig/Extension/EmojiExtension.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Extension;
+
+use Symfony\Component\Emoji\EmojiTransliterator;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+final class EmojiExtension extends AbstractExtension
+{
+    private static array $transliterators = [];
+
+    public function __construct()
+    {
+        if (!class_exists(EmojiTransliterator::class)) {
+            throw new \LogicException('You cannot use the "emojify" filter as the "Emoji" component is not installed. Try running "composer require symfony/emoji".');
```

Comment:
(geniuine curiosity) This could theorically never happen in "real situation" as the extension is re-register if the class does not exist...  (and i saw you use LogicException for that // and DX will be better "if" this happen)

My question is more: how to decide when we must add sugar on Logic things like that and when not ?

Reply:
For people using the bridge without the the framework bundle

Reply:
It seems obvious right now but ... yeah ok it was probably when i asked too :)  thx